bandc_c+a.rb

```diff
@@ -1,27 +1,43 @@
 puts "Welcome! Select classic or advanced game! For classic press 'c', for advanced press 'a'."
 game_type = gets.chomp
 
-while game_type == "c" or "a"
+def classic_check(variable)
+  if variable.size != 4
+    puts "Please, try again. Your guess must contain exactly 4 digits!"
+  end
+        
+  if variable.uniq.length != 4
+    puts "Please, try again. Your guess must contain 4 unique digits!"
+  end
+    
+  if variable.include?(0)
+    puts "Please, try again. You cannot use zero in your guess!"
+  end
+end
+
+def advanced_check(variable)
+  if variable.size != 6
+    puts "Please, try again. Your guess must contain exactly 6 digits!"
+  end
+        
+  if variable.uniq.length != 6
+    puts "Please, try again. Your guess must contain 6 unique digits!"
+  end
+end
```

Comment:
Almost same function as classic_check. Try use parameter for game type, and may be hash with   details of game: size, flags with needed checks, etc.